core/web/eth_keys_controller.go

```diff
@@ -136,13 +136,13 @@ func (ekc *ETHKeysController) Create(c *gin.Context) {
 		return
 	}
 
-	key, err := ethKeyStore.Create(chain.ID())
+	key, err := ethKeyStore.Create(c.Request.Context(), chain.ID())
 	if err != nil {
 		jsonAPIError(c, http.StatusInternalServerError, err)
 		return
 	}
 
-	state, err := ethKeyStore.GetState(key.ID(), chain.ID())
+	state, err := ethKeyStore.GetState(c, key.ID(), chain.ID())
```

Comment:
Why is `c` passed in some places but `c.Request.Context()` in others?

Reply:
I'll cleanup - based on https://github.com/gin-gonic/gin/blob/ecdbbbe9483dd12222f2085f717a2c7cb5ac55fe/context.go#L1180 I *should* just be able to pass `c`. However, it's not clear to me how key/values get set on the context in the middleware (i.e. I can't find invocations of `WithValue` from the go tooling Context or something resembling `Set` from *gin.Context). I'll timebox for now and just use `c.Request.Context()` everywhere.